src/commands/pull.js

```diff
@@ -0,0 +1,98 @@
+const mkdirp = require('mkdirp');
+const fs = require('fs');
+const path = require('path');
+const tiged = require('tiged');
+const AdmZip = require('adm-zip');
+const semverGt = require('semver/functions/gt');
+const { get } = require('axios');
+const { ApifyCommand } = require('../lib/apify_command');
+const { success, error } = require('../lib/outputs');
+const { getLoggedClientOrThrow } = require('../lib/utils');
+
+class PullCommand extends ApifyCommand {
+    async run() {
+        const { args } = this.parse(PullCommand);
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const { actorId } = args;
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+        const { name, versions } = actor;
+
+        const latestVersion = versions.reduce((match, curr) => {
+            if (semverGt(`${curr.versionNumber}.0`, `${match.versionNumber}.0`)) return curr;
+            return match;
+        });
+
+        const dirpath = path.join(cwd, name);
+        mkdirp.sync(dirpath);
+
+        if (!(fs.readdirSync(dirpath).length === 0)) {
+            error(`Directory ${dirpath} is not empty. Please empty it or choose another directory.`);
+        }
+
+        let isPullSuccessful = false;
+
+        switch (latestVersion.sourceType) {
+            case 'TARBALL': {
+                const { tarballUrl } = latestVersion;
+
+                const { data } = await get(tarballUrl, { responseType: 'arraybuffer' });
+
+                const zipFile = new AdmZip(Buffer.from(data, 'binary'));
+                zipFile.extractAllTo(dirpath);
+                break;
+            }
+            case 'SOURCE_FILES': {
+                const { sourceFiles } = latestVersion;
+                for (const file of sourceFiles) {
+                    if (!file.folder) {
+                        mkdirp.sync(dirpath);
+                        fs.writeFileSync(`${dirpath}/${file.name}`, file.content);
```

Comment:
This should also check for the `file.format`, it can also be `BASE64`, in which case the `file.content` needs to be parsed.